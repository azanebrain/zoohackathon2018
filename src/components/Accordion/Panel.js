import styled from 'styled-components';

const Panel = styled.div`
background-color: white;
display: block;
max-height: ${props => (!props.isActive ? 0 : '500px')};
overflow: hidden;
padding: 0;
transition: max-height 0.25s linear;

.link-more { display: none; }
`;

const PanelContainer = styled.div`
padding: 0 16px 16px 84px;
font-size: 14px;
`;

export { Panel, PanelContainer };
