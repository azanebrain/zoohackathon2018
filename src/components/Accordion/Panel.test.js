import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Panel } from './Panel';

test('Active Panel renders 500px tall', () => {
  const tree = renderer.create(<Panel isActive />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('max-height', '500px');
});

test('Inactive Panel renders 0px tall', () => {
  const tree = renderer.create(<Panel isActive={false} />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('max-height', '0');
});
