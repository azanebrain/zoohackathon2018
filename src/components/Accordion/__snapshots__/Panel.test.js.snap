// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active Panel renders 500px tall 1`] = `
.c0 {
  background-color: white;
  display: block;
  max-height: 500px;
  overflow: hidden;
  padding: 0;
  -webkit-transition: max-height 0.25s linear;
  transition: max-height 0.25s linear;
}

.c0 .link-more {
  display: none;
}

<div
  className="c0"
/>
`;

exports[`Inactive Panel renders 0px tall 1`] = `
.c0 {
  background-color: white;
  display: block;
  max-height: 0;
  overflow: hidden;
  padding: 0;
  -webkit-transition: max-height 0.25s linear;
  transition: max-height 0.25s linear;
}

.c0 .link-more {
  display: none;
}

<div
  className="c0"
/>
`;
